### ISSUE TITLE
[Documentation] Logging section

### ISSUE THREAD

**Issue by sylvlecl** [powsybl/pypowsybl#391]
* **Do you want to request a *feature* or report a *bug*?**

Doc

* **What is the expected behavior?**

We should add a section about logging, in the user guides of the documentation.
In particular we need to mention:
- the logger name
- the mechanism to get TRACE level logs
- maybe a note about performances, because a large amount of logs (for ex. when importing large cases) has a real impact on the

**Comment by geofjamg** [comment]
I have documented the TRACE level in my PR.
Regarging java_ extra variables, the problem if you define a formatter with this variable is that if another log is sent (maybe by another library) and for sure without this variable set, we endup with a lot of exception coming from the formatter. We would need to avoid that a feature added only in Python 3.10 to define a default value for missing variables( https://docs.python.org/3/library/logging.html#logging.Formatter): "Changed in version 3.10: The defaults parameter was added."
Only portable workaround is to define a custom handler....
So maybe we should not document this to avoid this pitfall?
